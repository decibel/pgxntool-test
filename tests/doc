#!/bin/bash

trap 'echo "ERROR: $BASH_SOURCE: line $LINENO" >&2' ERR
set -o errexit -o errtrace -o pipefail
#set -o xtrace -o verbose

if [ "$1" == "-v" ]; then
  verboseout=1
  shift
fi
BASEDIR=`cd ${0%/*}; pwd`

. $BASEDIR/../.env
. $TOPDIR/lib.sh
cd $TEST_DIR

get_html() {
  local rc
  html=`echo doc/*.html`
  [ "$html" != 'doc/*.html' ] || html=''
  html=$(for f in $html; do echo $f; done) # This allows for easy grepping
  if [ -n "$other_html" ]; then
    debug_vars 9 html other_html
    rc=0
    echo "$html" | grep -v $other_html || rc=$?
    if [ $rc -gt 1 ]; then
      error "grep returned $rc"
      exit 2
    fi
  else
    debug_vars 9 html
    echo "$html"
  fi
}

check_html() {
  html="$1"
  expected="$2"
  [ $# -eq 2 ] || die 3 "Wrong # of args for check_html()"

  [ "$html" == "$expected" ] || die 5 "make install did not produce expected documentation (expected '$expected', got '$html')"
}

which asciidoc &>/dev/null || which asciidoctor &>/dev/null || die 2 "asciidoc or asciidoctor not found"

rsync -a --delete repo/ doc_repo
cd doc_repo
input=`ls doc/*.adoc doc/*.asciidoc`
expected=$(echo "$input" | sed -Ee 's/(adoc|asciidoc)$/html/')

rm -f $expected
other_html=`get_html`
debug_vars 1 input expected other_html

out Make with no ASCIIDOC should not create docs
ASCIIDOC='' make install

[ -z "`get_html`" ] || die 5 "ASCIIDOC='' make install produced the following .html files in doc: " `get_html`

out Make sure missing ASCIIDOC errors out
ASCIIDOC='' make html || rc=$?
if [ $rc != 0 ]; then
  out make returned $rc
else
  die 5 "ASCIIDOC='' make html did not fail"
fi

out Make sure make install with ASCIIDOC works
make install
html=`get_html`
check_html "$html" "$expected"

out Make sure make clean does not clean docs
make clean
html2=`get_html`
[ "$html" == "$html2" ] || die 6 "make clean changed .html files (from '$html' to '$html2')"

out Make sure make docclean cleans docs
make print-ASCIIDOC_HTML
make print-DOCS_HTML
make docclean
[ -z "`get_html`" ] || die 5 "make docclean left html files behind: " `get_html`

out "Test ASCIIDOC_EXTS='asc'"
ASCIIDOC_EXTS='asc' make html
check_html "`get_html`" 'doc/adoc_doc.html
doc/asc_doc.html
doc/asciidoc_doc.html'

ASCIIDOC_EXTS='asc' make docclean
[ -z "`get_html`" ] || die 5 "ASCIIDOC_EXTS='asc' make docclean left html files behind: " `get_html`

check_log

# vi: expandtab sw=2 ts=2
