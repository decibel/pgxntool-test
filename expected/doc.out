# Make with no ASCIIDOC should not create docs
cp sql/pgxntool-test.sql sql/pgxntool-test--0.1.0.sql
@INSTALL@ -c -d '@PG_LOCATION@/share/extension'
@INSTALL@ -c -d '@PG_LOCATION@/share/doc/extension'
@INSTALL@ -c -m 644 ./sql/pgxntool-test--0.1.0.sql ./sql/pgxntool-test--0.1.0--0.1.1.sql ./pgxntool-test.control  '@PG_LOCATION@/share/extension/'
@INSTALL@ -c -m 644 ./doc/* '@PG_LOCATION@/share/doc/extension/'
# Make sure missing ASCIIDOC errors out
pgxntool/base.mk:125: Could not find "asciidoc" or "asciidoctor". Add one of them to your PATH,
pgxntool/base.mk:125: or set ASCIIDOC to the correct location.
pgxntool/base.mk:125: *** Could not build %doc/adoc_doc.html.  Stop.
# make returned 2
# Make sure make install with ASCIIDOC works
@INSTALL@ -c -d '@PG_LOCATION@/share/extension'
@INSTALL@ -c -d '@PG_LOCATION@/share/doc/extension'
/opt/local/bin/asciidoctor  doc/adoc_doc.adoc
/opt/local/bin/asciidoctor  doc/asciidoc_doc.asciidoc
@INSTALL@ -c -m 644 ./sql/pgxntool-test--0.1.0.sql ./sql/pgxntool-test--0.1.0--0.1.1.sql ./pgxntool-test.control  '@PG_LOCATION@/share/extension/'
@INSTALL@ -c -m 644 ./doc/* '@PG_LOCATION@/share/doc/extension/'
# Make sure make clean does not clean docs
rm -rf ../distribution_test-0.1.0.zip sql/pgxntool-test--0.1.0.sql
rm -rf results/ regression.diffs regression.out tmp_check/ log/
# Make sure make docclean cleans docs
ASCIIDOC_HTML is recursive variable set to "doc/adoc_doc.html doc/asciidoc_doc.html"
DOCS_HTML is recursive variable set to "doc/adoc_doc.html doc/asciidoc_doc.html"
rm -f doc/adoc_doc.html doc/asciidoc_doc.html
# Test ASCIIDOC_EXTS='asc'
/opt/local/bin/asciidoctor  doc/asc_doc.asc
/opt/local/bin/asciidoctor  doc/adoc_doc.adoc
/opt/local/bin/asciidoctor  doc/asciidoc_doc.asciidoc
rm -f doc/asc_doc.html doc/adoc_doc.html doc/asciidoc_doc.html
